src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -411,25 +390,14 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
         let ty_res = match ty_res {
             Err(()) | Ok(Res::Err) => {
                 return if ns == Namespace::ValueNS {
-                    self.variant_field(path_str, current_item, module_id, extra_fragment)
+                    self.variant_field(path_str, current_item, module_id)
                 } else {
-                    // See if it only broke because of the namespace.
```

Comment:
Hmm, I need to double check this didn't impact anything - `resolution_failure` doesn't look in different namespaces. There might not be tests for this path?

Reply:
Ok I lied, it does look in different namespaces: https://github.com/rust-lang/rust/blob/66a9cfadf89439ad3ea1b67b996c0a90ca4763bf/src/librustdoc/passes/collect_intra_doc_links.rs#L1569-L1572
So this is handled properly:
```
warning: unresolved link to `file::into_iter`
 --> tmp.rs:1:6
  |
1 | /// [type@file::into_iter]
  |      ^^^^^^^^^^^^^^^^^^^^ `file` is a macro, not a module or type, and cannot have associated items
```
What's going on is it only looks at other namespaces _if there are multiple path segments_. So https://github.com/rust-lang/rust/blob/66a9cfadf89439ad3ea1b67b996c0a90ca4763bf/src/librustdoc/passes/collect_intra_doc_links.rs#L1114 and https://github.com/rust-lang/rust/blob/66a9cfadf89439ad3ea1b67b996c0a90ca4763bf/src/librustdoc/passes/collect_intra_doc_links.rs#L1242-L1244 are still necessary, because they handle cases like `value@file` and `type@file`. I could probably clean that up at some point, but it's not causing any bugs so I think I'll leave it be for now.

Reply:
Please file an issue so we don't forget.

Reply:
Opened https://github.com/rust-lang/rust/issues/77150